src/services/PortfolioService.ts

```
@@ -0,0 +1,361 @@
+
+    const currentAvailable = Number.parseFloat(balance.available)
+    const currentAverageCost = Number.parseFloat(balance.averageCost)
+    const transactionAmount = Number.parseFloat(transaction.amount)
+    const transactionPrice = transaction.price ? Number.parseFloat(transaction.price) : 0
+
+    switch (transaction.type) {
+      case TransactionType.BUY:
+      case TransactionType.DEPOSIT:
+      case TransactionType.TRANSFER_IN:
+        // Increase balance and update average cost
+        const newTotal = currentAvailable + transactionAmount
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)